### PR TITLE
iometer_windows: New case to test multiple disks

### PR DESCRIPTION
**Pull request by YongxueHong**
New a case to test the multiple disks at the same time.

ID: 1566862
Signed-off-by: Yongxue Hong yhong@redhat.com

### CODE REVIEW

File: generic/tests/cfg/iometer_windows.cfg [hunk 1]
```diff
@@ -73,3 +73,37 @@
                     bg_mode = yes
                 - after_test:
                     bg_mode = no
+        - with_multi_scsi_disks:
+            only virtio_scsi
+            run_timeout = 3600
+            icf_name = "iometer_multi_disks.icf"
+            format_multi_disks = yes
+            disk_indexes = "1 2 3 4 5 6"
```
Comment (peixiu) [inline generic/tests/cfg/iometer_windows.cfg]: hi yhong, why here test 6 disks? any other requirement exist? I see the polarion case test 4 disks~

File: generic/tests/cfg/iometer_windows.cfg [hunk 2]
```diff
@@ -73,3 +73,37 @@
...
+            disk_letters = "I J K L M N"
+            disk_fstypes = "fat32 fat ntfs fat32 fat ntfs"
```
Comment (peixiu) [inline generic/tests/cfg/iometer_windows.cfg]: also here why repeat each file system types? and missed type Res~

File: generic/tests/cfg/iometer_windows.cfg (hunk 1 above)
Comment (YongxueHong) [reply generic/tests/cfg/iometer_windows.cfg]: ok, will update it, if you think 4 disks are ok for yours testing.

File: generic/tests/cfg/iometer_windows.cfg (hunk 2 above)
Comment (YongxueHong) [reply generic/tests/cfg/iometer_windows.cfg]: ok, will it by the manual case.

Comment (peixiu) [inline generic/tests/cfg/iometer_windows.cfg]: Hi yhong,

The detail file system type support status for windows os as follows:
1. On Win2016-64, Win2019-64: NTFS, FAT32, ReFS, test 3 disks is enough.
2. On Win2012-64, Win2012-r2: NTFS, FAT, FAT32, ReFS, need test 4 disks.
3. On others, Win7-32/64, Win8-32/64, Win8.1-32/64, Win10-32/64, Win2008-32/64/r2: NTFS, FAT32, only 2 disks is enough.

Please help to re-configure the cfg to match the os supported status in actually. Thanks~

Comment (peixiu) [reply generic/tests/cfg/iometer_windows.cfg]: Hi yhong, Update the detail FS support status on windows:

1. On win2012-64, win2012-r2, win2016-64, win2019-64, supported file system type as: FAT, FAT32, NTFS, ReFS.
2. On Win7-32/64, Win8-32/64, Win8.1-32/64, Win10-32/64, Win2008-32/64/r2, supported file system type as: FAT, FAT32, NTFS.

Note: FAT system does support on  <=3G partition, ReFS does support on  >=1.5G partition, FAT32 does support on <=128G partition, and NTFS does support on <=2T partition.

So, suggest tested with 2G disk for Refs file system, others file system keep 1G also is ok.

Best Regrds~
Peixiu

Comment (YongxueHong) [reply generic/tests/cfg/iometer_windows.cfg]: > Hi yhong, Update the detail FS support status on windows:
> 
>     1. On win2012-64, win2012-r2, win2016-64, win2019-64, supported file system type as: FAT, FAT32, NTFS, ReFS.
> 
>     2. On Win7-32/64, Win8-32/64, Win8.1-32/64, Win10-32/64, Win2008-32/64/r2, supported file system type as: FAT, FAT32, NTFS.
> 
> 
> Note: FAT system does support on <=3G partition, ReFS does support on >=1.5G partition, FAT32 does support on <=128G partition, and NTFS does support on <=2T partition.
> 
> So, suggest tested with 2G disk for Refs file system, others file system keep 1G also is ok.
> 
> Best Regrds~
> Peixiu

Hi @peixiu  thanks for you comments. it's great help to me.

**Comment by YongxueHong** [comment]
hi， @peixiu 
Please help me to review it, thanks.

File: generic/tests/cfg/iometer_windows.cfg [hunk 3]
```diff
@@ -73,3 +73,52 @@
...
+                disk_letters = "I J K"
+                disk_fstypes = "ntfs fat32 refs"
+                images += " stg0 stg1 stg2"
+                force_create_image_stg0 = yes
+                force_create_image_stg1 = yes
+                force_create_image_stg2 = yes
+                remove_image_stg0 = yes
+                remove_image_stg1 = yes
+                remove_image_stg2 = yes
+            Win2012, Win2012..r2:
```
Comment (peixiu) [inline generic/tests/cfg/iometer_windows.cfg]: Hi yhong, please update based that I updated support list secondly yesterday! 
    On win2012-64, win2012-r2, win2016-64, win2019-64, supported file system type as: FAT, FAT32, NTFS, ReFS. need to test 4 disk. each disk match a file system type.
    On Win7-32/64, Win8-32/64, Win8.1-32/64, Win10-32/64, Win2008-32/64/r2, supported file system type as: FAT, FAT32, NTFS. need to test 3 disk, each disk match with a file system type.
thus only need 2 iometer_**.icf file.
And suggest set all disk size to 2G, it match all 4 file system.

Comment (YongxueHong) [reply generic/tests/cfg/iometer_windows.cfg]: ok, got it, will update it, thanks.

**Comment by YongxueHong** [comment]
Hi, @peixiu 
Please help me review it, thanks.

**Comment by peixiu** [comment]
git am error:
[root@dell-per440-02 tests]# git am 2009.patch 
Applying: iometer: Add iometer's icfs to test multiple disks
.git/rebase-apply/patch:14: trailing whitespace.
Version 2006.07.27 
.git/rebase-apply/patch:17: trailing whitespace.
	
.git/rebase-apply/patch:330: trailing whitespace.
Version 2006.07.27 
.git/rebase-apply/patch:337: trailing whitespace.
Version 2006.07.27 
.git/rebase-apply/patch:340: trailing whitespace.
	
warning: squelched 1 whitespace error
warning: 6 lines add whitespace errors.
Applying: iometer_windows: New case to test multiple disks

File: generic/tests/cfg/iometer_windows.cfg [hunk 4]
```diff
@@ -73,3 +73,36 @@
...
+            remove_image_stg0 = yes
+            remove_image_stg1 = yes
+            remove_image_stg2 = yes
+            Win7, Win8, Win8..1, Win10, Win2008r2:
+                icf_name = "iometer_multi_3disks.icf"
+                disk_indexes = "1 2 3"
+                disk_letters = "I J K"
+                disk_fstypes = "ntfs fat32 fat"
+                images += " stg0 stg1 stg2"
+            Win2012, Win2012..r2, Win2016, Win2019:
```
Comment (peixiu) [inline generic/tests/cfg/iometer_windows.cfg]: can write to "Win2012, Win2016, Win2019:", Win2012 include Win2012-64 and Win2012-r2.

File: generic/tests/cfg/iometer_windows.cfg [hunk 5]
```diff
@@ -73,3 +73,36 @@
...
+            image_size_stg1 = 2G
+            image_size_stg2 = 2G
+            image_size_stg3 = 2G
+            force_create_image_stg0 = yes
+            force_create_image_stg1 = yes
+            force_create_image_stg2 = yes
+            remove_image_stg0 = yes
+            remove_image_stg1 = yes
+            remove_image_stg2 = yes
+            Win7, Win8, Win8..1, Win10, Win2008r2:
```
Comment (peixiu) [inline generic/tests/cfg/iometer_windows.cfg]: if these configurate can be set to default value? include all guests except win2012, win2016, win2019.

File: generic/tests/cfg/iometer_windows.cfg (hunk 4 above)
Comment (YongxueHong) [reply generic/tests/cfg/iometer_windows.cfg]: ok, thanks, update it.

File: generic/tests/cfg/iometer_windows.cfg (hunk 5 above)
Comment (YongxueHong) [reply generic/tests/cfg/iometer_windows.cfg]: ok, will update it, thanks.

File: generic/tests/cfg/iometer_windows.cfg [hunk 6]
```diff
@@ -73,3 +73,35 @@
                     bg_mode = yes
                 - after_test:
                     bg_mode = no
+        - with_multi_scsi_disks:
+            only virtio_scsi
+            run_timeout = 3600
+            format_multi_disks = yes
+            images = "image1"
```
Comment (peixiu) [inline generic/tests/cfg/iometer_windows.cfg]: could this line can be deleted? it seems no need add here, the default system image is image1~

**Comment by YongxueHong** [comment]
@peixiu  @lolyu pls help me to review, thanks.

File: generic/tests/cfg/iometer_windows.cfg (hunk 6 above)
Comment (peixiu) [reply generic/tests/cfg/iometer_windows.cfg]: that's ok~

**Comment by peixiu** [comment]
Tested on win2019 and win8.1-32 guest, all passed.
 (1/1) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win8.i386.1.io-github-autotest-qemu.iometer_windows.with_multi_scsi_disks.q35: PASS (1028.79 s)
 (1/1) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.iometer_windows.with_multi_scsi_disks.q35: PASS (1415.64 s)

Acked-by: Peixiu Hou phou@redhat.com

**Pull request merged**